types/recommendationrequest.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019 The MayaData Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CStorPoolRecommendationRequest is a kubernetes custom
+// resource that represents configuration input or requirement
+// to create or scale one CStor pool.
+type CStorPoolRecommendationRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec CStorPoolRecommendationRequestSpec `json:"spec"`
+}
+
+// CStorPoolRecommendationRequestSpec contains input or requirement
+// to create or scale one CStor pool.
+type CStorPoolRecommendationRequestSpec struct {
+	// PoolClusterName represents name of the cspc.
+	PoolClusterName string `json:"poolClusterName"`
+	// PoolClusterNamespace represents namespace of the cspc.
+	PoolClusterNamespace string `json:"poolClusterNamespace"`
+	// Capacity represents requested capacity for cspc
+	Capacity int64 `json:"capacity"`
+	// PoolType represents type of pool
+	PoolType string `json:"poolType"`
+	// GroupDetails represents no of parity and data disk
+	// This is required for raidz configuration.
+	GroupDetails `json:",inline"`
+	// Depending on this we will decide we want to create one cspc
+	// Or we want to perform expending of existing cspc
+	CStorPoolCluster *Reference `json:"cStorPoolCluster"`
+}
+
+// GroupDetails contains no of data and parity device count
+type GroupDetails struct {
+	// No of Data disk
```

Comment:
Can we put appropriate comments. e.g Minimum number of data disks that participate in forming a single cstor pool instance. 
Let us be very clear in our comments if the field impacts the pool instance or overall cluster & so on.